src/ucp/core/ucp_context.h

```diff
@@ -324,11 +324,11 @@ typedef struct ucp_tl_md {
 
 
 typedef struct ucp_context_alloc_md_index {
-    int            initialized;
+    int               initialized;
     /* Index of memory domain that is used to allocate memory of the given type
      * using ucp_memh_alloc(). */
-    ucp_md_index_t   md_index;
-    ucs_sys_device_t sys_dev;
+    ucp_md_index_t    md_index;
+    ucp_memory_info_t mem_info;
```

Comment:
do we really have to change this field? we can return mem_type from the user param as before..